### DISCUSSION TITLE
jdbc配置集群时url的写法

### DISCUSSION THREAD

**Discussion by yuhuafeng1109** [taosdata/TDengine#4641] [Q&A]
单个数据库时jdbc配置url写法如下：
spring.datasource.url=jdbc:TAOS://172.16.5.71:6030/songjing
那么创建数据库集群后，jdbc的url应该怎么正确配置？

**Comment by sangshuduo** [comment]
没有区别，只是变成可以连集群中任意服务器

**Reply by yuhuafeng1109** [reply]
我的意思是其中一台服务器挂了以后，怎么自动链接另外的服务器。
比如：我现在创建了两台服务器作为集群（172.16.5.239:6030，172.16.5.240:6030）,然后项目中配置了spring.datasource.url=jdbc:TAOS://172.16.5.239:6030/songjing，那么当172.16.5.239这台服务器挂了以后是不能自动链接到172.16.5.240这台上的。

**Comment by yuhuafeng1109** [comment]
我的意思是其中一台服务器挂了以后，怎么自动链接另外的服务器。
比如：我现在创建了两台服务器作为集群（172.16.5.239:6030，172.16.5.240:6030）,然后项目中配置了spring.datasource.url=jdbc:TAOS://172.16.5.239:6030/songjing，那么当172.16.5.239这台服务器挂了以后是不能自动链接到172.16.5.240这台上的。

**Comment by yuhuafeng1109** [comment]
@sangshuduo 在官网中有提到当使用JDBC连接TDengine集群时，可以使用客户端配置文件，在客户端配置文件中指定集群的firstEp、secondEp参数，这个方法试过是可以的，但是对项目部署来说比较麻烦，必须要taos.cfg配置文件才行。还有如果集群的服务器数量多的话，是依次类推吗，thirdEp、fourthEp。。。。

**Reply by matchmen** [reply]
JDBC-RESFUL支持吗？我试了，不行啊。配置为：jdbc:TAOS-RS://:/nacos_user?timezone=UTC-8&charset=UTF-8&cfgdir=taos

taos是taos.cfg的位置

**Comment by yuhuafeng1109** [comment]
mysql集群，jdbc url写法:
jdbc:mysql://[host:port],[host:port].../[database][?propertyName1][=propertyValue1][&propertyName2][=propertyValue2]...
TDengine没有类似mysql集群的url写法吗？
最好是可以支持spring.datasource.url=jdbc:TAOS://172.16.5.239:6030,/172.16.5.240:6030,/172.16.5.241:6030/songjing这样的用法。

**Comment by sangshuduo** [comment]
目前没有
